### PR TITLE
added OData support for agents and agencies booking

### PR DESCRIPTION
**Pull request by DFSko**

### CODE REVIEW

File: Api/Services/Accommodations/Bookings/BookingRecordsManager.cs [hunk 1]
```diff
@@ -130,15 +130,37 @@ static List<BookedRoom> UpdateSupplierReferenceCodes(List<BookedRoom> existingRo
...
+                    },
+                    Data = new SlimAccommodationBookingInfo
+                    {
+                        Id = booking.Id,
+                        ReferenceCode = booking.ReferenceCode,
+                        AccommodationName = booking.AccommodationName,
+                        CountryName = booking.Location.Country,
+                        LocalityName = booking.Location.Locality,
+                        Deadline = booking.DeadlineDate,
+                        Price = new MoneyAmount(booking.TotalPrice, booking.Currency),
```
Comment (mingazhev) [inline Api/Services/Accommodations/Bookings/BookingRecordsManager.cs:155]: Would the price filter and sort work in this case?
Maybe we should use primitive types here

**Review by mingazhev — APPROVED** [review]
Looks fine for me

File: Api/Services/Accommodations/Bookings/BookingRecordsManager.cs (hunk 1 above)
Comment (DFSko) [reply Api/Services/Accommodations/Bookings/BookingRecordsManager.cs]: Yes, sorting not work with this type, replaced to primitives

**Review by mingazhev — APPROVED** [review]

**Comment by mingazhev** [comment]
Need to discuss model changes with frontend

**Review by mingazhev — APPROVED** [review]
Looks ok, let's wait for @kirillta

File: Api/Models/Bookings/SlimAccommodationBookingInfo.cs [hunk 2]
```diff
@@ -22,40 +18,41 @@ public SlimAccommodationBookingInfo(Booking bookingInfo)
...
+        public int Id { get; init; }
+
+        public string ReferenceCode { get; init; }
 
-        public BookingStatuses Status { get; }
+        public BookingStatuses Status { get; init; }
+        public decimal Price { get; init; }
 
-        public MoneyAmount Price { get; }
+        public Currencies Currency { get; init; }
```
Comment (kirillta) [inline Api/Models/Bookings/SlimAccommodationBookingInfo.cs:38]: Why? We agreed to use `MoneyAmount` across the system

Comment (DFSko) [reply Api/Models/Bookings/SlimAccommodationBookingInfo.cs:38]: Because sorting and filtering not working on complex types

**Pull request merged**